packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -22,25 +22,21 @@ import {
   gridVisibleColumnFieldsSelector,
 } from '../columns';
 
-// Fixes https://github.com/mui/mui-x/issues/10056
-const globalScope = (typeof window === 'undefined' ? globalThis : window) as any;
-const evalCode = globalScope[atob('ZXZhbA==')] as <T>(source: string) => T;
+let hasEval: boolean;
 
-let hasEval: boolean | undefined;
-
-const getHasEval = () => {
+function getHasEval() {
   if (hasEval !== undefined) {
     return hasEval;
   }
 
   try {
-    hasEval = evalCode<boolean>('true');
+    hasEval = new Function('return true') as any;
```

Comment:
This is assigning the function to `hasEval` instead of assigning a boolean value to it. Maybe something like this instead?

```js
new Function('');
hasEval = true;

// or

hasEval = new Function('return true')() as boolean;
```

Reply:
True, I would be afraid with the first version that optimization tools remove `new Function('');` as side effect free.